kafka-commons/src/test/java/uk/gov/homeoffice/digital/sas/kafka/producer/KafkaProducerServiceTest.java

```diff
@@ -37,6 +51,9 @@ class KafkaProducerServiceTest {
   @Mock
   private KafkaTemplate<String, KafkaEventMessage<Profile>> kafkaTemplate;
 
+  @Mock
```

Comment:
remove @Mock annotation as this is in test sometimes as a mock and one time as a spy.
Or create a separate variable for the spy (in this case you won't need the assignments `..=spy()/mock() inside the test methods)

Reply:
I have removed @Mock. Also had to add `responseFuture = mock(CompletableFuture.class);`
into `sendMessage_actionOnResource_messageIsSentWithCorrectArguments` Test.